artman/cli/main2.py

```diff
@@ -136,13 +136,15 @@ def parse_args(*args):
         help='[Optional] Directory to store output generated by artman. '
         'Default to `./artman-genfiles`', )
     parser.add_argument(
-        '--input-dir',
+        '--root-dir',
         type=str,
         default='',
         help='[Optional] Directory with all input that is needed by artman, '
-             'which include but not limited to API protos, service config '
-             'yaml, and GAPIC config yaml. Default to the settings from user '
-             'configuration.',
+        'which include but not limited to API protos, service config yaml '
+        'and GAPIC config yaml. It will be passed to protobuf compiler via '
+        '`-I` flag in order to generate descriptor. If not specified, it '
+        'will first look up in artman user config. If not found, an error '
```

Comment:
"will first look up in the artman config" -> "will derive the root directory from the source proto files"

Reply:
I didn't implement that feature here. Unlike google apis protos, there is no guarantee it will be correct.

Reply:
I am not sure I understood your plan. Do you plan to put this logic in g3artman?

I think it makes sense to implement that feature in artman, not g3artman. It then becomes an artman convention to structure your proto source directories according to your proto packages.

Reply:
Per offline discussion, considering this resolved, since using such logic relies on potentially unstable assumptions about the directory structure of your source proto path.